pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -847,7 +899,7 @@ func (nc *Controller) processTaintBaseEviction(node *v1.Node, observedReadyCondi
 	}
 }
 
-func (nc *Controller) processNoTaintBaseEviction(node *v1.Node, observedReadyCondition *v1.NodeCondition, gracePeriod time.Duration) {
+func (nc *Controller) processNoTaintBaseEviction(node *v1.Node, observedReadyCondition *v1.NodeCondition, gracePeriod time.Duration, pods []*v1.Pod) {
```

Comment:
Can you move the Pod pointer part to a separate PR?

Reply:
moved to #83780